docs/userguide/expectations.md

```diff
@@ -0,0 +1,78 @@
+# Concept of expectation and matcher 
+
+Validation of the code under test (the tested logic of procedure/function etc.) is performed by comparing the actual data against the expected data.
+To do that we use concept of expectation and a matcher to perform the check on the data.
+
+It's best to give an example to get an idea what is what
+```sql
+begin
+  ut.expect( 'the tested value' ).to_( equal('the expected value') );
```

Comment:
The `to_()` is a base syntax for extensible matchers and the `to_equal()` is just a wrapper.
Not all matcher have wrappers for them and getting developers familiar with the `to_` syntax seemed like a good idea.